deployment/ccip/shared/stateview/state.go

```diff
@@ -95,6 +96,36 @@ type CCIPOnChainState struct {
 	Chains      map[uint64]evm.CCIPChainState
 	SolChains   map[uint64]solana.CCIPChainState
 	AptosChains map[uint64]aptosstate.CCIPChainState
+	evmMu       *sync.RWMutex
+}
+
+func (c CCIPOnChainState) EVMChains() []uint64 {
```

Comment:
```suggestion
func (c *CCIPOnChainState) EVMChains() []uint64 {
```
Changing this to pointer will ensure there is only one mutex per state object. Passing by value may lead to subtle race bugs. Better to apply for all four methods EVMChains, EVMChainState, MustGetEVMChainState and WriteEVMChainState

Reply:
it will create issues with having methods bothcon pointer and direct object. Idt it will create any race as the mutex itself is a pointer